frontend/www/js/omegaup/components/course/Statistics.vue

```diff
@@ -0,0 +1,192 @@
+import T from '../../lang';
+import * as ui from '../../ui';
+@Component({
+  components: {
+    highcharts: Chart,
+  },
+})
+export default class Statistics extends Vue {
+  @Prop() course!: types.CourseDetails;
+  @Prop() problemStats!: types.CourseProblemStatistics[];
+  T = T;
+  //chart options
+  selected = this.varianceChartOptions;
+  options = [
+    { value: this.varianceChartOptions, text: T.courseStatisticsVariance },
+    { value: this.averageChartOptions, text: T.courseStatisticsAverageScore },
+    {
+      value: this.highScoreChartOptions,
+      text: T.courseStatisticsStudentsAbove,
+    },
+    {
+      value: this.lowScoreChartOptions,
+      text: T.courseStatisticsStudentsScored,
+    },
+    { value: this.minimumChartOptions, text: T.courseStatisticsMinimumScore },
+    { value: this.maximumChartOptions, text: T.courseStatisticsMaximumScore },
+  ];
+  //get chart options
+  get varianceChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsScoreVariance,
+      '{y}',
+      T.courseStatisticsVariance,
+      this.getStatistic('variance'),
+      this.maxVariance,
+      this.problems,
+    );
+  }
+  get averageChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsAverageScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('average'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  get highScoreChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsStudentsAbove,
+      '{y} %',
+      T.wordsPercentage,
+      this.getStatistic('high_score_percentage'),
+      100,
+      this.problems,
+    );
+  }
+  get lowScoreChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsStudentsScored,
+      '{y} %',
+      T.wordsPercentage,
+      this.getStatistic('low_score_percentage'),
+      100,
+      this.problems,
+    );
+  }
+  get minimumChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsMinimumScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('minimum'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  get maximumChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsMaximumScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('maximum'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  //helper functions
+  get problems() {
+    return this.problemStats.map(
+      (problem) => `${problem.assignment_alias} - ${problem.problem_alias}`,
+    );
+  }
+  get maxPoints() {
+    let maxPoints = 0;
+    for (const problem of this.problemStats) {
+      if (problem.maxPoints > maxPoints) maxPoints = problem.maxPoints;
+    }
+    return maxPoints;
+  }
+  get maxVariance() {
+    let maxVariance = 0;
+    const variance = this.getStatistic('variance');
+    for (let i = 0; i < variance.length; i++) {
+      if (variance[i] > maxVariance) maxVariance = variance[i];
+    }
```

Comment:
```suggestion
    for (const variance of this.getStatistic('variance')) {
      if (variance > maxVariance) maxVariance = variance;
    }
```

this can also use `for...of`!